compiler/rustc_hir_analysis/src/hir_ty_lowering/dyn_trait.rs

```diff
@@ -28,11 +34,24 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
         hir_id: hir::HirId,
         hir_bounds: &[hir::PolyTraitRef<'tcx>],
         lifetime: &hir::Lifetime,
-        representation: DynKind,
+        syntax: TraitObjectSyntax,
     ) -> Ty<'tcx> {
         let tcx = self.tcx();
         let dummy_self = tcx.types.trait_object_dummy_self;
 
+        match syntax {
```

Comment:
took me a while to understand that this match got moved out of `prohibit_or_lint_bare_trait_object_ty` and where the code from `compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs` went but this makes things a lot neater.